MobiFlight/MobiFlightFirmwareUpdater.cs

```diff
@@ -34,66 +34,89 @@ public static bool IsValidFirmwareFilepath(string filepath)
 
         public static bool Update(MobiFlightModule module)
         {
-            if (module.Board.AvrDudeSettings == null)
+            string firmwareName;
+
+            if (module.Board.AvrDudeSettings != null)
+            {
+                firmwareName = module.Board.AvrDudeSettings.GetFirmwareName(module.Board.Info.LatestFirmwareVersion);
+            }
+            else if (module.Board.UsbDriveSettings != null)
+            {
+                firmwareName = module.Board.UsbDriveSettings.GetFirmwareName(module.Board.Info.LatestFirmwareVersion);
+            }
+            else
             {
                 Log.Instance.log($"Firmware update requested for {module.Board.Info.MobiFlightType} ({module.Port}) however no update settings were specified in the board definition file. Module update skipped.", LogSeverity.Error);
                 return false;
             }
 
-            var FirmwareName = module.Board.AvrDudeSettings.GetFirmwareName(module.Board.Info.LatestFirmwareVersion);
-            return UpdateFirmware(module, FirmwareName);
+            return UpdateFirmware(module, firmwareName);
         }
 
         public static bool Reset(MobiFlightModule module)
         {
-            if (module.Board.AvrDudeSettings == null)
+            if (String.IsNullOrEmpty(module.Board.Info.ResetFirmwareFile))
             {
                 Log.Instance.log($"Firmware reset requested for {module.Board.Info.MobiFlightType} ({module.Port}) however no reset settings were specified in the board definition file. Module reset skipped.", LogSeverity.Error);
                 return false;
             }
 
-            var FirmwareName = module.Board.AvrDudeSettings.ResetFirmwareFile;
-            return UpdateFirmware(module, FirmwareName);
+            return UpdateFirmware(module, module.Board.Info.ResetFirmwareFile);
         }
 
         public static bool UpdateFirmware(MobiFlightModule module, String FirmwareName)

```

Comment:
Changes to this method are best viewed in side-by-side mode instead of inline mode.